python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,144 @@
+                                          defaultTemplates={"refDataset": "gaia_dr2_20200414"}):
+
+    catalog = pipeBase.connectionTypes.Input(
+        doc="Source table in parquet format, per visit",
+        dimensions=("instrument", "visit", "band"),
+        storageClass="DataFrame",
+        name="sourceTable_visit",
+        deferLoad=True
+    )
+
+    refcat = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="Reference catalog",
+        name="{refDataset}",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True
+    )
+
+    measurement = pipeBase.connectionTypes.Output(
+        doc="Per-detector measurement",
+        dimensions=("instrument", "visit", "detector", "band"),
+        storageClass="MetricValue",
+        name="metricvalue_{package}_{metric}"
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if config.connections.refDataset == '':
+            self.prerequisiteInputs.remove("refcat")
+
+
+class DetectorTableMeasurementConfig(CatalogMeasurementBaseTaskConfig,
+                                     pipelineConnections=DetectorTableMeasurementConnections):
+    """Configuration for DetectorTableMeasurementTask."""
+
+    columns = pexConfig.ListField(doc="Columns from sourceTable_visit to load.",
+                                  dtype=str, default=['coord_ra', 'coord_dec', 'detector'])
+
+    refObjLoader = pexConfig.ConfigurableField(
+        target=LoadReferenceCatalogTask,
+        doc="Reference object loader",
+    )
+
+    def setDefaults(self):
+        self.refObjLoader.refObjLoader.ref_dataset_name = "gaia_dr2_20200414"
+        self.refObjLoader.refObjLoader.requireProperMotion = True
+        self.refObjLoader.refObjLoader.anyFilterMapsToThis = 'phot_g_mean'
+        self.refObjLoader.doApplyColorTerms = False
+
+    def validate(self):
+        super().validate()
+        if 'detector' not in self.columns:
+            msg = "The column `detector` must be appear in the list of columns."
+            raise pexConfig.FieldValidationError(DetectorTableMeasurementConfig.columns, self, msg)
+        if self.connections.refDataset != self.refObjLoader.refObjLoader.ref_dataset_name:
+            msg = "The reference datasets specified in connections and reference object loader must match."
+            raise pexConfig.FieldValidationError(DetectorTableMeasurementConfig.columns, self, msg)
+
+
+class DetectorTableMeasurementTask(CatalogMeasurementBaseTask):
+    """Base class for science performance metrics measured on single-dector source catalogs."""
+
+    ConfigClass = DetectorTableMeasurementConfig
+    _DefaultName = "detectorTableMeasurementTask"
+
+    def run(self, catalog, refcat, refcatCalib):
+        return self.measure.run(self.config.connections.metric, catalog,
+                                refcat=refcat, refcatCalib=refcatCalib)
```

Comment:
You will need to unify your `run` signatures, since the one you use in your example pipeline above `NumSourcesTask` has quite a different call signature, so it will fail to work here. Is there any `run` args uniformity between the various measurement tasks in faro?

I also see `NumSourcesTask` and `NumSourcesMergeTask` having quite different `run` signatures (and no docstrings!). If those are supposed to be interchangeable, they need to all have the same `run`.

Reply:
Now using the run method inherited from `CatalogMeasurementBaseTask`. Using keyword arguments to handle situation that metrics will require different input arguments.